audio-accuracy-test/audio_accuracy_test.py

```diff
@@ -43,82 +74,145 @@ def close(self):
 
 class AudioTester(object):
     def __init__(self, samp_rate):
+        print  # Pad debug messages
         self.ww_recognizer = LocalRecognizer(samp_rate, 'en-us')
         self.listener = ResponsiveRecognizer(self.ww_recognizer)
+        print
         speech_logger.setLevel(100)  # Disables logging to clean output
 
     def test_audio(self, file_name):
         source = FileMockMicrophone(file_name)
         ee = pyee.EventEmitter()
 
         class SharedData:
-            found = False
+            times_found = 0
 
         def on_found_wake_word():
-            SharedData.found = True
+            SharedData.times_found += 1
 
         ee.on('recognizer_loop:record_begin', on_found_wake_word)
 
         try:
-            self.listener.listen(source, ee)
+            while True:
+                self.listener.listen(source, ee)
         except EOFError:
             pass
 
-        return SharedData.found
+        return SharedData.times_found
 
 
-BOLD = '\033[1m'
-NORMAL = '\033[0m'
-GREEN = '\033[92m'
-RED = '\033[91m'
+class Color:
+    BOLD = '\033[1m'
+    NORMAL = '\033[0m'
+    GREEN = '\033[92m'
+    RED = '\033[91m'
 
 
 def bold_str(val):
-    return BOLD + str(val) + NORMAL
+    return Color.BOLD + str(val) + Color.NORMAL
 
 
 def get_root_dir():
     return dirname(dirname(__file__))
 
-if __name__ == "__main__":
-    directory = join('audio-accuracy-test', 'data', 'query_after')
-    query = join(directory, '*.wav')
+
+def get_file_names(folder):
+    query = join(folder, '*.wav')
     root_dir = get_root_dir()
     full_path = join(root_dir, query)
     file_names = sorted(glob(full_path))
 
+    if len(file_names) < 1:
+        raise IOError
+
+    return file_names
+
+
+def test_audio_files(tester, file_names, on_file_finish):
     num_found = 0
-    total = len(file_names)
+    for file_name in file_names:
+        short_name = os.path.basename(file_name)
+        times_found = tester.test_audio(file_name)
+
+        num_found += times_found
+        on_file_finish(short_name, times_found)
+
+    return num_found
+
+
+def file_frame_rate(file_name):
+    wf = wave.open(file_name, 'rb')
+    frame_rate = wf.getframerate()
+    wf.close()
+    return frame_rate
 
-    if total < 1:
-        print(bold_str(RED+"Error: No wav files found in " + directory))
-        exit(1)
+
+def test_false_negative(directory):
+    file_names = get_file_names(directory)
 
     # Grab audio format info from first file
-    ex_file = wave.open(file_names[0], 'rb')
+    tester = AudioTester(file_frame_rate(file_names[0]))
+
+    def print_status(word, short_name):
+        print("Wake word " + bold_str(word) + " - " + short_name)
```

Comment:
There is no real advantage of declaring this inner method to print those data. Replace this method to the `print` function directly where they are needed. However, if you have a reusable logic across all your unit tests, you should move this method outside and use it in all unit tests methods instead.